plugins/agent.py

```
@@ -282,6 +282,45 @@ def _make_run_config(cfg: dict, bot, backend: str) -> RunConfig:
+        return suffix[: max_bytes]
+
+    # Strategy: try to find a word boundary (space) within the budget.
+    # Scan backwards from budget to find last space before the cutoff.
+    cut_at = budget
+    for i in range(budget, max(0, budget - 200), -1):
+        if encoded[i - 1] == 0x20:  # space byte in UTF-8
+            cut_at = i - 1
+            break
+
+    # Now ensure cut_at doesn't land inside a multi-byte UTF-8 sequence.
+    # UTF-8 continuation bytes have the pattern 10xxxxxx (0x80–0xBF).
```

Ruff (0.15.15)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)